plugin/pkg/admission/podtolerationrestriction/admission.go

```diff
@@ -135,12 +136,13 @@ func (p *Plugin) Validate(ctx context.Context, a admission.Attributes, o admissi
 		// fall back to cluster's whitelist of tolerations.
 		if whitelist == nil {
 			whitelist = p.pluginConfig.Whitelist
+			whitelistScope = "cluster"
 		}
 
 		if len(whitelist) > 0 {
 			// check if the merged pod tolerations satisfy its namespace whitelist
 			if !tolerations.VerifyAgainstWhitelist(pod.Spec.Tolerations, whitelist) {
-				return fmt.Errorf("pod tolerations (possibly merged with namespace default tolerations) conflict with its namespace whitelist")
```

Comment:
does this read clearer?

`pod tolerations conflict with its %s whiltelist`

it does not feel necessary to add the possibly merged bit if we can attribute it back to the proper scope.

Reply:
As far as I understood, the admission controller does the following
```
allTolerations = merge(PodTolerations, NamespaceTolerations)
if namespaceWhitelist:
checkWhitelist(allTolerations,namespaceWhitelist)
else:
checkWhitelist(allTolerations,clusterWhitelist)
```

"possibly merged" refers to the tolerations, not the whitelist used.